src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -951,9 +951,9 @@ fn preprocess_link<'a>(
     }
 
     // Parse and strip the disambiguator from the link, if present.
-    let (path_str, disambiguator) = match Disambiguator::from_str(&link) {
-        Ok(Some((d, path))) => (path.trim(), Some(d)),
-        Ok(None) => (link.trim(), None),
+    let (link_text, path_str, disambiguator) = match Disambiguator::from_str(&link) {
+        Ok(Some((d, path, link_text))) => (link_text, path.trim(), Some(d)),
+        Ok(None) => (link.trim().to_owned(), link.trim(), None),
```

Comment:
It seems confusing to have the order of the function different from the order of the match.
```suggestion
    let (disambiguator, path_str, link_text) = match Disambiguator::from_str(&link) {
        Ok(Some((d, path, link_text))) => (Some(d), path.trim(), link_text),
        Ok(None) => (None, link.trim(), link.trim().to_owned()),
```